### PR TITLE
end to end

### PR DESCRIPTION
**Pull request by SamSteu**
Here is the PR including the end to end tests for Milestone. There are 4 end to end files in total : 
- `AroundYouFlowEndToEnd.kt` which contains the user flow of the around you screen except the AI part : 
     - click on a profile
     - test the behavior of the dialogs to send requests without sending it
     - filter screen and check  it's influence on the around you screen
- `SettingsFlowEndToEnd.kt` contains the user flow of the setting screen and all of it's features except the deletion of the account : 
     - check if our profile is well displayed
     - modify our profile and check that it has been updated
     - test the block functionality
     - test the already met functionality
     - test the toggle discoverability feature
     - test the report
     - test the log out
- `ReceiveRequestFlowEndToEnd.kt` contains multiple end to end tests that check the behavior of the meeting request system when you receive a request : 
     - receive a request and then a cancellation message TIME, CLOSED, DISTANCE or CANCELLED and check everything is updated
     - receives a request and decline it
     - receive a request and accept it
- `SendRequestFlowEndToEnd.kt` contains multiple end to end tests that check the behavior of the meeting request system when you send a request except the time cancellation because this feature is disabled for the moment 
     - send a request and then receive a cancellation message  CLOSED, DISTANCE or CANCELLED and check everything is updated
     - send a request and then cancel your request
     - send a request and receive a positive response
     - send a request and receive a negative response

I also renamed `MainActivity` to `Icebreakrr`

Here is the jacoco test report, we have 82% coverage on the project overall : 

image

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    }
+  }
+
+  @Test
+  fun receiveRequestTimeCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      receiveRequestAndCheckProfile()
+
+      // receives a cancellation for time :
+      message.addData("message", "TIME").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:89]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    }
+  }
+
+  @Test
+  fun receiveRequestDistanceCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      receiveRequestAndCheckProfile()
+
+      // receives a cancellation for distance :
+      message.addData("message", "DISTANCE").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:109]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    }
+  }
+
+  @Test
+  fun receiveRequestCancelledCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      receiveRequestAndCheckProfile()
+
+      // receives a cancellation because the other one cancelled :
+      message.addData("message", "CANCELLED").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:129]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    }
+  }
+
+  @Test
+  fun receiveRequestClosedCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      receiveRequestAndCheckProfile()
+
+      // receives a cancellation because the other one cancelled :
+      message.addData("message", "CLOSED").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:149]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/SendRequestFlowEndToEnd.kt [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    }
+  }
+
+  @Test
+  fun sendRequestDistanceCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      sendRequestAliceAndCheck()
+
+      // receives a cancellation for time :
+      message.addData("message", "DISTANCE").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/SendRequestFlowEndToEnd.kt:112]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/SendRequestFlowEndToEnd.kt [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    }
+  }
+
+  @Test
+  fun sendRequestCancelledCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      sendRequestAliceAndCheck()
+
+      // receives a cancellation for time :
+      message.addData("message", "CANCELLED").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/SendRequestFlowEndToEnd.kt:129]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/SendRequestFlowEndToEnd.kt [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+    }
+  }
+
+  @Test
+  fun sendRequestClosedCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      sendRequestAliceAndCheck()
+
+      // receives a cancellation for time :
+      message.addData("message", "CLOSED").addData("title", "MEETING CANCELLATION")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/SendRequestFlowEndToEnd.kt:146]: Nitpick: Hardcoded string

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+  private fun receiveRequestAndCheckProfile() {
+    // check if everything is displayed in the around you
+    composeTestRule.onNodeWithTag("aroundYouScreen").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("topBar").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("filterButton").assertIsDisplayed().assertHasClickAction()
+    composeTestRule.onNodeWithTag("bottomNavigationMenu").assertIsDisplayed()
+
+    // receives a message
+    message
+        .addData("title", "MEETING REQUEST")
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:204]: "Hey, do you want to meet?

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+      // check that alice disapear :
+      composeTestRule.onNodeWithTag("profileCard").assertIsNotDisplayed()
+      composeTestRule
+          .onNodeWithTag("badgeNotification", useUnmergedTree = true)
+          .assertIsNotDisplayed()
+      scenario.close()
+    }
+  }
+
+  private fun receiveRequestAndCheckProfile() {
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:195]: Nitpick: maybe add kdoc to this function, to make the code more readable.

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+      composeTestRule.onNodeWithTag("profileCard").assertIsNotDisplayed()
+      composeTestRule
+          .onNodeWithTag("badgeNotification", useUnmergedTree = true)
+          .assertIsNotDisplayed()
+      scenario.close()
+    }
+  }
+
+  @Test
+  fun receiveRequestDistanceCancellationFlowEndToEnd() {
```
Comment (arthur-mrgt) [inline app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:104]: Nitpick: The name is lengthy and not entirely clear. Change it or add kdoc for better code clarity.

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt [hunk 11]
```diff
@@ -55,19 +55,23 @@ class MeetingRequestService : FirebaseMessagingService() {
...
-          appProcess.processName == packageName) {
-        return true
+      for (appProcess in appProcesses) {
+        if (appProcess.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND &&
+            appProcess.processName == packageName) {
+          return true
+        }
       }
+    } catch (e: NullPointerException) {
+      Log.d("endToEnd", "Normal in testing mode but this should not happen in production.")
```
Comment (arthur-mrgt) [inline app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:74]: Question: I don't really understand what does this do.

**Review by arthur-mrgt — APPROVED** [review]
Great work on this test suite! It provides comprehensive coverage of key scenarios and demonstrates excellent use of modern tools like Hilt. The reusable helper methods and structured approach show clear attention to detail and maintainability.

A few areas for improvement:

Hardcoded Strings: Using constants or resources instead of inline strings (e.g., "MEETING RESPONSE") would enhance maintainability.

Test Naming: Test names could be more descriptive and concise, clearly stating the action and expected outcome (e.g., test_meetingRequestDeclined_hidesProfile).

Documentation: Adding brief comments to describe the scenarios would improve readability for future reviews or collaborators.

Overall, this is an impressive and solid test suite—great job! 🎉

File: app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt (hunk 10 above)
Comment (SamSteu) [reply app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt:104]: I added a _and_ since it is literally receiving a request, then receiving a distance cancellation and check that everything is coherent

File: app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt (hunk 11 above)
Comment (SamSteu) [reply app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt:74]: In Andoid tests, the context inside the app is `null`, so the call to `Context.ACTIVITY_SERVICE` will throw a `NullPointerException`. Since in the end-to-end tests I don't test if I have received a notification, I decided to skip this edge case that way, and since the context should never be `null` in production this should be okay. There are many more elegant ways to do this but it is the easiest way.

**Review by arthur-mrgt — APPROVED** [review]
Lgtm (the final one)
